src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/SaveAsCsvFileStreamWriter.cs

```diff
@@ -48,23 +48,57 @@ public SaveAsCsvFileStreamWriter(Stream stream, SaveResultsAsCsvRequestParams re
         /// </param>
         public override void WriteRow(IList<DbCellValue> row, IList<DbColumnWrapper> columns)
         {
-            string delimiter = ",";
+            char delimiter = ',';
             if(!string.IsNullOrEmpty(saveParams.Delimiter))
             {
-                delimiter = saveParams.Delimiter;
-            }            
+                // first char in string
+                delimiter = saveParams.Delimiter[0];
+            }
+
+            string lineSeperator = Environment.NewLine;
+            if(!string.IsNullOrEmpty(saveParams.LineSeperator))
+            {
+                lineSeperator = saveParams.LineSeperator;
+            }
+
+            char textIdentifier = '"';
+            if(!string.IsNullOrEmpty(saveParams.TextIdentifier))
+            {
+                // first char in string
+                textIdentifier = saveParams.TextIdentifier[0];
+            }
+
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+            int codepage;
+            Encoding encoding;
+            try
+            {
+                if(int.TryParse(saveParams.Encoding, out codepage))
+                {
+                    encoding = Encoding.GetEncoding(codepage);
+                }
+                else
+                {
+                    encoding = Encoding.GetEncoding(saveParams.Encoding);
```

Comment:
will this throw in the most common case where `Encoding` isn't provided?  If so, you may consider rewriting to first null check `Encoding` and set `encoding = Encoding.GetEncoding("utf-8");` in the normal flow control (either in the `else` or as a default value when declaring `encoding`.

Basically, I like to avoid 1st-chance handled exceptions when possible since I often run the debugger to break on all exceptions which works best when exceptions are only used for real error conditions.